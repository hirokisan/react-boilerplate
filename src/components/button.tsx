import React from 'react';

export interface ButtonProps {
  children?: React.ReactNode;
  onClick?: (e: unknown) => void;
}
const styles = {
  border: '1px solid #eee',
  borderRadius: 3,
  backgroundColor: '#FFFFFF',
  cursor: 'pointer',
  fontSize: 15,
  padding: '3px 10px',
  margin: 10
};

const Button: React.SFC<ButtonProps> = props => (
  <button onClick={props.onClick} style={styles} type="button">
    {props.children}
  </button>
);
Button.defaultProps = {
  children: null,
  onClick: () => {}
};
export default Button;
