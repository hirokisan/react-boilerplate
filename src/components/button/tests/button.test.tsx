import { render } from '@testing-library/react';
import React from 'react';

import { Button } from '../button';

import serializer from 'jest-emotion';

expect.addSnapshotSerializer(serializer);

describe('Button', () => {
  it('Renders with text', () => {
    const { container } = render(<Button content="My Button" />);
    expect(container).toMatchSnapshot();
  });
});
