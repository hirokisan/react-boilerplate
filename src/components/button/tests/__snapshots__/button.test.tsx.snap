// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renders with text 1`] = `
.emotion-0 {
  border: 1px solid #eee;
  border-radius: 3px;
  background-color: white;
  cursor: pointer;
  font-size: 15px;
  padding: 3px 10px;
  margin: 10px;
}

<div>
  <div />
  <button
    class="emotion-0"
  >
    My Button
  </button>
  <div>
    subTitle
  </div>
</div>
`;
